packages/react-dom/src/client/ReactDOMFiberComponent.js

```diff
@@ -312,7 +311,7 @@ function setInitialDOMProperties(
     } else if (propKey === AUTOFOCUS) {
       // We polyfill it separately on the client during commit.
       // We blacklist it here rather than in the property list because we emit it in SSR.
-    } else if (registrationNameModules.hasOwnProperty(propKey)) {
+    } else if (propKey[0] === 'o' && propKey[1] === 'n') {
```

Comment:
It might be worth extracting this into a function like `isEventName` so it minifies better.